src/Command/Ingester/Native/NativeUserIngesterCommand.php

```diff
@@ -240,26 +302,39 @@ private function getAvailableStartIndex(): int
         return $index + 1;
     }
 
+    /**
+     * @throws NonUniqueResultException
+     * @throws NoResultException
+     */
+    private function getAvailableAssignmentStartIndex(): int
+    {
+        $index = $this->assignmentRepository
+            ->createQueryBuilder('a')
+            ->select('MAX(a.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+
+        return $index + 1;
+    }
+
     private function executeNativeInsertions(ResultSetMapping $mapping): void
     {
-        try {
-            if (!empty($this->userQueryParts) && !empty($this->assignmentQueryParts) && !$this->isDryRun) {
-                $userQuery = sprintf(
-                    'INSERT INTO users (id, username, password, roles, group_id) VALUES %s',
-                    implode(',', $this->userQueryParts)
-                );
+        if (!empty($this->userQueryParts) && !$this->isDryRun) {
+            $userQuery = sprintf(
+                'INSERT INTO users (id, username, password, roles, group_id) VALUES %s',
```

Comment:
Is there a possibility to create a repository class for these queries?

Reply:
Good point 👍 , actually this is already done as part of this PR: https://github.com/oat-sa/simple-roster/pull/120/files#diff-bcebdc86fff68e52e416becf7ae40334R31 , so eventually this whole command will be reworked, now it's kind of messy and it does way too many things.